statusHandler/termuic/termuiRenders/widgetsRender.go

```diff
@@ -0,0 +1,252 @@
+
+	consensusRoundTime := wr.getFromCacheAsUint64(core.MetricRoundTime)
+	rows = fmt.Sprintf("%sConsensus round time: %ds\n", rows, consensusRoundTime)
+
+	rows = fmt.Sprintf("%s\n", rows)
+
+	numLiveValidators := wr.getFromCacheAsUint64(core.MetricLiveValidatorNodes)
+	rows = fmt.Sprintf("%sLive validator nodes: %v\n", rows, numLiveValidators)
+
+	numConnectedNodes := wr.getFromCacheAsUint64(core.MetricConnectedNodes)
+	rows = fmt.Sprintf("%sNetwork connected nodes: %v\n", rows, numConnectedNodes)
+
+	numConnectedPeers := wr.getFromCacheAsUint64(core.MetricNumConnectedPeers)
+	rows = fmt.Sprintf("%sThis node is connected to %v peers\n", rows, numConnectedPeers)
+
+	return "Chain info", rows
+}
+
+func (wr *WidgetsRender) prepareListWithLogsForDisplay(logData []string) {
+	wr.lLog.Title = "Log info"
+	wr.lLog.TextStyle = ui.NewStyle(ui.ColorWhite)
+	wr.lLog.Rows = wr.prepareLogLines(logData, wr.lLog.Size().Y)
+	wr.lLog.WrapText = true
+	return
+}
+
+//TODO duplicate code next pull request refactor
+func (wr *WidgetsRender) prepareLogLines(logData []string, size int) []string {
+	logDataLen := len(logData)
+
+	if logDataLen > size {
+		return logData[logDataLen-size : logDataLen]
+	}
+	return logData
+}
+
+func (wr *WidgetsRender) prepareLoads() {
+	cpuLoadPercentI, _ := wr.termuiConsoleMetrics.Load(core.MetricCpuLoadPercent)
+	cpuLoadPercent := cpuLoadPercentI.(uint64)
+	wr.cpuLoad.Title = "CPU Load"
+	wr.cpuLoad.Percent = int(cpuLoadPercent)
+
+	memLoadPercentI, _ := wr.termuiConsoleMetrics.Load(core.MetricMemLoadPercent)
+	memLoadPercent := memLoadPercentI.(uint64)
+	memTotalMemoryBytesObj, _ := wr.termuiConsoleMetrics.Load(core.MetricTotalMem)
+	memTotalMemoryBytes := memTotalMemoryBytesObj.(uint64)
+	wr.memoryLoad.Title = "Memory load"
+	wr.memoryLoad.Percent = int(memLoadPercent)
+	wr.memoryLoad.Label = fmt.Sprintf("%d%% / T: %s", memLoadPercent, core.ConvertBytes(memTotalMemoryBytes))
+
+	recvLoad := wr.getFromCacheAsUint64(core.MetricNetworkRecvPercent)
+	recvBps := wr.getFromCacheAsUint64(core.MetricNetworkRecvBps)
+	recvBpsPeak := wr.getFromCacheAsUint64(core.MetricNetworkRecvBpsPeak)
+	wr.networkRecv.Title = "Network - recv:"
+	wr.networkRecv.Percent = int(recvLoad)
+	wr.networkRecv.Label = fmt.Sprintf("%d%% / rate: %s/s / peak rate: %s/s",
+		recvLoad, core.ConvertBytes(recvBps), core.ConvertBytes(recvBpsPeak))
+
+	sentLoad := wr.getFromCacheAsUint64(core.MetricNetworkSentPercent)
+	sentBps := wr.getFromCacheAsUint64(core.MetricNetworkSentBps)
+	sentBpsPeak := wr.getFromCacheAsUint64(core.MetricNetworkSentBpsPeak)
+	wr.networkSent.Title = "Network - sent:"
+	wr.networkSent.Percent = int(sentLoad)
+	wr.networkSent.Label = fmt.Sprintf("%d%% / rate: %s/s / peak rate: %s/s",
+		sentLoad, core.ConvertBytes(sentBps), core.ConvertBytes(sentBpsPeak))
+}
+
+func (wr *WidgetsRender) getFromCacheAsUint64(metric string) uint64 {
+	val, ok := wr.termuiConsoleMetrics.Load(metric)
+	if !ok {
+		return math.MaxUint64
```

Comment:
Why return `MaxUint64` in case of error and not 0?

Reply:
To be able to see that something went wrong. 0 could mean uninitialized.

Reply:
Ok, no problem.